packages/legend-extension-dsl-persistence/src/models/protocols/pure/v1/transformation/pureProtocol/V1_DSLPersistence_ProtocolHelper.ts

```diff
@@ -0,0 +1,855 @@
+    return serialize(V1_appendOnlyModelSchema, protocol);
+  }
+  throw new UnsupportedOperationError(
+    `Unable to serialize batch mode`,
+    protocol,
+  );
+};
+
+const V1_deserializeIngestMode = (
+  json: PlainObject<V1_IngestMode>,
+): V1_IngestMode => {
+  switch (json._type) {
+    case V1_BatchModeType.NONTEMPORAL_SNAPSHOT:
+      return deserialize(V1_nontemporalSnapshotModelSchema, json);
+    case V1_BatchModeType.UNITEMPORAL_SNAPSHOT:
+      return deserialize(V1_unitemporalSnapshotModelSchema, json);
+    case V1_BatchModeType.BITEMPORAL_SNAPSHOT:
+      return deserialize(V1_bitemporalSnapshotModelSchema, json);
+    case V1_BatchModeType.NONTEMPORAL_DELTA:
+      return deserialize(V1_nontemporalDeltaModelSchema, json);
+    case V1_BatchModeType.UNITEMPORAL_DELTA:
+      return deserialize(V1_unitemporalDeltaModelSchema, json);
+    case V1_BatchModeType.BITEMPORAL_DELTA:
+      return deserialize(V1_bitemporalDeltaModelSchema, json);
+    case V1_BatchModeType.APPEND_ONLY:
+      return deserialize(V1_appendOnlyModelSchema, json);
+    default:
+      throw new UnsupportedOperationError(
+        `Unable to deserialize batch mode '${json._type}'`,
+      );
+  }
+};
+
+// merge strategy
+
+enum V1_MergeStrategyType {
+  NO_DELETES_MERGE_STRATEGY = 'noDeletesMergeStrategy',
+  DELETE_INDICATOR_MERGE_STRATEGY = 'deleteIndicatorMergeStrategy',
+  OPAQUE_MERGE_STRATEGY = 'opaqueMergeStrategy',
+}
+
+const V1_noDeletesMergeStrategyModelSchema = createModelSchema(
+  V1_NoDeletesMergeStrategy,
+  {
+    _type: usingConstantValueSchema(
+      V1_MergeStrategyType.NO_DELETES_MERGE_STRATEGY,
+    ),
+  },
+);
+
+const V1_deleteIndicatorMergeStrategyModelSchema = createModelSchema(
+  V1_DeleteIndicatorMergeStrategy,
+  {
+    _type: usingConstantValueSchema(
+      V1_MergeStrategyType.DELETE_INDICATOR_MERGE_STRATEGY,
+    ),
+    deleteProperty: primitive(),
+    deleteValues: list(primitive()),
+  },
+);
+
+const V1_serializeMergeStrategy = (
+  protocol: V1_MergeStrategy,
+): PlainObject<V1_MergeStrategy> => {
+  if (protocol instanceof V1_NoDeletesMergeStrategy) {
+    return serialize(V1_noDeletesMergeStrategyModelSchema, protocol);
+  } else if (protocol instanceof V1_DeleteIndicatorMergeStrategy) {
+    return serialize(V1_deleteIndicatorMergeStrategyModelSchema, protocol);
+  }
+  throw new UnsupportedOperationError(
+    `Unable to serialize merge strategy`,
+    protocol,
+  );
+};
+
+const V1_deserializeMergeStrategy = (
+  json: PlainObject<V1_MergeStrategy>,
+): V1_MergeStrategy => {
+  switch (json._type) {
+    case V1_MergeStrategyType.NO_DELETES_MERGE_STRATEGY:
+      return deserialize(V1_noDeletesMergeStrategyModelSchema, json);
+    case V1_MergeStrategyType.DELETE_INDICATOR_MERGE_STRATEGY:
+      return deserialize(V1_deleteIndicatorMergeStrategyModelSchema, json);
+    default:
+      throw new UnsupportedOperationError(
+        `Unable to deserialize merge strategy '${json._type}'`,
```

Comment:
Let's update the message format to start with `Can't`

Reply:
Fixed in #990